numpy/lib/function_base.py

```diff
@@ -4351,14 +4357,27 @@ def meshgrid(*xi, copy=True, sparse=False, indexing='xy'):
     `meshgrid` is very useful to evaluate functions on a grid.
 
     >>> import matplotlib.pyplot as plt
-    >>> x = np.arange(-5, 5, 0.1)
-    >>> y = np.arange(-5, 5, 0.1)
+    >>> x = np.linspace(-5, 5, 101)
+    >>> y = np.linspace(-5, 5, 101)
     >>> xx, yy = np.meshgrid(x, y, sparse=True)
     >>> z = np.sin(xx**2 + yy**2) / (xx**2 + yy**2)
     >>> h = plt.contourf(x, y, z)
     >>> plt.axis('scaled')
     >>> plt.show()
 
+    If the function depends on all coordinates, as in the example above, you
+    can use the parameter ``sparse=True`` to save memory and computation time.
+    Here, the sparse coordinate grid reduces the memory from 2*101*101 = 20.402
+    to 101+101 = 202 values.  Likewise, the expression ``xx**2 + yy**2`` is
+    reduced from 20.401 square operations to only 202 square operations, but
```

Comment:
I don't think this is correct. There's both the squaring and addition operations to consider: doesn't the addition require 101**2 operations?

In any case, my two cents is that this could benefit from less detail as it's easy to lose the forest through the trees here. I'd advocate for chopping out the text from `Here,`: IMO the best way to provide more detail would be to add a reference to the [2011 NumPy paper](https://arxiv.org/abs/1102.1523), which has an excellent detailed example of broadcasting with "sparse" arrays.

Just my two cents though - leaving the detail is also fine, but we should double check to make sure the numbers are correct.

Reply:
I was only referring to the square operations. I anticipated that the square operation was more expensive than addition, but that doesn't seem to be the case. So the overall gain is theoretically just a factor 3 (beacuse the sparse sqares can be neglected compared to the addition). A bit more testing also shows that a real gain only appears when x, y have order of 1000 elements or larger, which I find surprisingly high.

![grafik](https://user-images.githubusercontent.com/2836374/132109337-3200ff9d-c2e6-4c8c-b258-b4f59d326882.png)

Anyway, it's not within my scope to discuss performance characteristics. I've removed everything from "Here" as suggested.